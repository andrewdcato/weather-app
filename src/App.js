import React, { useState } from 'react';
import Divider from './components/Divider';
import Layout from './components/Layout';
import LocationInput from './components/LocationInput';
import WeatherDisplay from './components/weather/WeatherDisplay';
import useGetForecast from './hooks/useGetForecast';

export default function App() {
  // Latitude and Longitude for weather API call
  const [lat, setLat] = useState(null);
  const [lng, setLng] = useState(null);

  // Forecast and animation toggle
  const {
    current,
    daily,
    hourly,
    isLoading,
  } = useGetForecast(lat, lng);

  return (
    <Layout>
      <LocationInput
        setLat={setLat}
        setLng={setLng}
      />

      <Divider />

      <WeatherDisplay
        current={current}
        daily={daily}
        hourly={hourly}
        isLoading={isLoading}
      />
    </Layout>

  );
}

