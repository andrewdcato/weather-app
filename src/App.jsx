import { useEffect, useState } from 'react';
import Divider from './components/Divider';
import LocationInput from './components/LocationInput';
import WeatherDisplay from './components/WeatherDisplay';

function useGetForecast(lat, lng) {
  // Loading animation toggle
  const [isLoading, setIsLoading] = useState(false);

  // Forecast data
  const [current, setCurrent] = useState(null);
  const [forecast, setForecast] = useState(null);

  useEffect(() => {
    const fetchData = async (lat, lng) => {
      if (!!lat && !!lng) {
        setIsLoading(true);

        const {
          REACT_APP_WEATHER_API_URL: API_URL,
          REACT_APP_WEATHER_API_KEY: API_KEY
        } = process.env

        const url = `${API_URL}?lat=${lat}&lon=${lng}&units=imperial&appid=${API_KEY}`;

        await fetch(url)
          .then(response => response.json())
          .then(result => {
            setCurrent(result?.current);
            setForecast(result?.daily);
          })
          .then(() => setIsLoading(false));
      }
    }

    fetchData(lat, lng);
  }, [lat, lng]);

  return { current, forecast, isLoading };
}

export default function App() {
  // Latitude and Longitude for weather API call
  const [lat, setLat] = useState(null);
  const [lng, setLng] = useState(null);

  // Forecast and animation toggle
  const {
    current,
    forecast,
    isLoading
  } = useGetForecast(lat, lng);

  return (
    <div className="mx-auto p-4 bg-blue-100 flex h-screen justify-center items-center">
      <div className="flex flex-wrap">
        <div className="w-full px-2">
          <div className="p-6 relative">
            <LocationInput setLat={setLat} setLng={setLng} />
          </div>

          <Divider />

          {(!!lat && !!lng) &&
            <WeatherDisplay
              current={current}
              forecast={forecast}
              isLoading={isLoading}
            />
          }
        </div>
      </div>
    </div>
  );
}

