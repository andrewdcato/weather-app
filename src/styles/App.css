body {
  background-color: white;
  color: darkslategrey;
}

/* @media (prefers-color-scheme: dark) {
  body {
    background-color: darkslategrey;
    color: white;
  }
} */

.app {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-top: 12%;
}

