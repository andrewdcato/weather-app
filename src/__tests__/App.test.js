import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from '../App';

test('renders location search field', () => {
  render(<App />);
  const inputElement = screen.getByTestId('location-input');

  expect(inputElement).toBeInTheDocument();
  expect(inputElement).toHaveAttribute('type', 'text');
});

// test('successfully finds a valid city', () => {
//   render(<App />);
//   const inputElement = screen.getByTestId('location-input');
//   userEvent.type('inputElement', 'New York, NY, USA');

//   expect(inputElement).toBeInTheDocument();
//   expect(inputElement).toHaveAttribute('type', 'text');
// })
