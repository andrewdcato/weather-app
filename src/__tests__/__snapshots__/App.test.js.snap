// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="h-screen px-12 py-12"
>
  <div
    className="w-full h-full"
  >
    <div
      className="w-full mb-2"
    >
      <div
        className="flex justify-center"
      >
        <label
          className="sr-only"
          htmlFor="city"
        >
          City
        </label>
        <input
          className="w-full block shadow-sm focus:ring-blue-500 focus:border-blue-500 border-gray-300 rounded-md"
          data-testid="location-input"
          name="city"
          onChange={[Function]}
          placeholder="Enter a City"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="block sm:flex justify-between items-center flex-wrap"
    />
    <div
      className="bg-blue-300 text-gray-800 rounded-lg overflow-hidden shadow my-6"
    >
      <div
        className="px-6 py-6 relative"
      >
        <div
          className="flex justify-center items-center"
        >
          Please enter and select a city above to see the weather.
        </div>
      </div>
    </div>
  </div>
</div>
`;
